data-model.d.ts

```diff
@@ -287,4 +287,18 @@ export interface DataFactory<OutQuad extends BaseQuad = Quad, InQuad extends Bas
      * @see Quad
      */
     quad(subject: InQuad['subject'], predicate: InQuad['predicate'], object: InQuad['object'], graph?: InQuad['graph']): OutQuad;
+
+    /**
+     * @param original The original term.
+     * @return A new instance of the term such that newTermInstance.equals(original) returns true.
+     * @see Term
+     */
+    fromTerm(original: Term): Term;
```

Comment:
```suggestion
    fromTerm<T extends Term>(original: T): T;
```

This way to ensure that we keep the original typing of the term

Reply:
The original term type needs to be kept indeed, but this doesn't need to be the precise subtype created by a specific data factory.

Something like this would be needed: https://github.com/rubensworks/rdf-data-factory.js/blob/master/lib/DataFactory.ts#L96-L102

Reply:
Of course! But maybe it's better with overloads. I updated my suggestion above.

With overloads, you get a little better resulting typing maybe?
<img width="862" alt="image" src="https://github.com/user-attachments/assets/beec14a3-b584-4ab7-937a-9b88a50db1fe">

Reply:
Also, since `Quad` is no also a term, do actually want a `fromTerm(q: Quad)` variant, or should that be limited to `fromQuad`?

Reply:
> Also, since Quad is no also a term, do actually want a fromTerm(q: Quad) variant

I guess that should be supported as well indeed.

Strictly speaking, we don't even need `fromQuad` anymore. But maybe we want to keep it for legacy and convenience reasons.